app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -109,21 +118,41 @@ public CommentsViewModel(final @NonNull Environment environment) {
 
     final Observable<List<Comment>> comments = paginator.paginatedData().share();
 
+    final Observable<Boolean> currentUserIsCreator = Observable.combineLatest(
+      currentUser,
+      project.map(Project::creator),
+      Pair::create
+    )
+      .map(userAndCreator -> userAndCreator.first != null && userAndCreator.first.id() == userAndCreator.second.id());
+
+    final Observable<Boolean> userCanComment = Observable.combineLatest(
+      currentUserIsCreator,
```

Comment:
yep can do, i think i had that moment of "hm thats a lot of `combineLatest`" but the code for some outputs read nicely

but they'll still read nicely